WordPress/gutenberg
# [Mobile] - Full-width/wide alignment support for more blocks

`Gutenberg Mobile PR` -> https://github.com/wordpress-mobile/gutenberg-mobile/pull/2605
`WordPress iOS PR` -> https://github.com/wordpress-mobile/WordPress-iOS/pull/14865
`WordPress Android PR` -> https://github.com/wordpress-mobile/WordPress-Android/pull/12916

## Description
This PR is a continuation of https://github.com/WordPress/gutenberg/pull/24598 it adds full-width/wide alignment for all blocks that support them **except** the Columns block which is being added in another PR.

**Blocks that support full-width/wide alignments**: Video, Latest-posts, Gallery, Media & text, Columns, Pullquote, Cover, Group, and Image. (Columns will be added soon as well)

## How has this been tested?


<details><summary>Steps to test</summary>
---

With metro running:

#### Video
- With a premium or self-hosted site
- Add a Video block
- Change the alignment of the block to `Full width`
- **Expect** to see the block edge-to-edge of the screen
- Change the alignment to `wide`
- **Expect** to see some margin between the block and the edges of the screen

#### Latest-posts
- Add a Latest-posts block
- Change the alignment of the block to `Full width`
- **Expect** to see the block edge-to-edge of the screen
- Change the alignment to `wide`
- **Expect** to see some margin between the block and the edges of the screen

#### Gallery
- Add a Gallery block, select some images
- Change the alignment of the block to `Full width`
- **Expect** to see the block edge-to-edge of the screen
- Change the alignment to `wide`
- **Expect** to see some margin between the block and the edges of the screen

#### Gallery within a Group block
- Add a Group block and change its alignment to `Full width`
- Within the Group block, a Gallery block, select some images
- Change the alignment of the block to `Full width`
- **Expect** to see the block edge-to-edge of the screen
- Change the alignment to `wide`
- **Expect** to see some margin between the block and the edges of the screen

</details>

## Screenshots 

<details><summary>Video, Latest-posts</summary>

iOS | Android
-|-
<kbd><img src="https://user-images.githubusercontent.com/4885740/94016989-229c6380-fdaf-11ea-8031-13d6d123e42d.png" width="250" /></kbd> | <kbd><img src="https://user-images.githubusercontent.com/4885740/94016993-24662700-fdaf-11ea-9216-3adef8386ba6.png" width="250" /></kbd> 

iPad | iOS Safe area
-|-
<kbd><img src="https://user-images.githubusercontent.com/4885740/94017190-6db67680-fdaf-11ea-8673-faa5dd57b320.png" width="250" /></kbd> | <kbd><img src="https://user-images.githubusercontent.com/4885740/94017193-6f803a00-fdaf-11ea-9cc7-3a99356eee5c.png" width="250" /></kbd> 

</details>

<details><summary>Gallery, Pullquote</summary>

iOS | Android
-|-
<kbd><img src="https://user-images.githubusercontent.com/4885740/94017495-ce45b380-fdaf-11ea-8472-eaf7e16a6a70.png" width="250" /></kbd> | <kbd><img src="https://user-images.githubusercontent.com/4885740/94017504-d00f7700-fdaf-11ea-8677-093f44b0faf6.png" width="250" /></kbd> 

iPad | iOS Safe area
-|-
<kbd><img src="https://user-images.githubusercontent.com/4885740/94017616-efa69f80-fdaf-11ea-86a5-0ea38607cbb6.png" width="250" /></kbd> | <kbd><img src="https://user-images.githubusercontent.com/4885740/94017598-ea495500-fdaf-11ea-94c0-d4779cdcded8.png" width="250" /></kbd> 

</details>

<details><summary>Media & Text</summary>

iOS | Android
-|-
<kbd><img src="https://user-images.githubusercontent.com/4885740/94017757-17960300-fdb0-11ea-84b5-59d033e56e84.png" width="250" /></kbd> | <kbd><img src="https://user-images.githubusercontent.com/4885740/94017766-195fc680-fdb0-11ea-99be-04901e85d98c.png" width="250" /></kbd> 

iPad | iOS Safe area
-|-
<kbd><img src="https://user-images.githubusercontent.com/4885740/94017822-28df0f80-fdb0-11ea-8479-0534ad4db4ad.png" width="250" /></kbd> | <kbd><img src="https://user-images.githubusercontent.com/4885740/94017831-2b416980-fdb0-11ea-9bbb-c7d8fbacc643.png" width="250" /></kbd> 

</details>




## Types of changes
New feature

## Checklist:
